src/blazingmq/_session.py

```diff
@@ -207,6 +221,190 @@ def __repr__(self) -> str:
+            emitted when the buffer reaches the high watermark value, and a
+            notification `.SlowConsumerNormal` is emitted when the buffer is
+            back to the low watermark.
+        stats_dump_interval:
+            The interval (in seconds) at which to dump stats into the logs.  If
+            0, disable the recurring dump of stats (final stats are always
+            dumped at the end of the session).  The default is 5min; the value
+            must be a multiple of 30s, in the range ``[0s - 60min]``.
+    """
+
+    def __init__(
+        self,
+        message_compression_algorithm: Optional[CompressionAlgorithmType] = None,
+        timeouts: Optional[Timeouts] = None,
+        host_health_monitor: Union[BasicHealthMonitor, None] = (DefaultMonitor()),
+        num_processing_threads: Optional[int] = None,
+        blob_buffer_size: Optional[int] = None,
+        channel_high_watermark: Optional[int] = None,
+        event_queue_watermarks: Optional[tuple[int, int]] = None,
+        stats_dump_interval: Optional[float] = None,
+    ) -> None:
+        self.message_compression_algorithm = message_compression_algorithm
+        self.timeouts = timeouts
+        self.host_health_monitor = host_health_monitor
+        self.num_processing_threads = num_processing_threads
+        self.blob_buffer_size = blob_buffer_size
+        self.channel_high_watermark = channel_high_watermark
+        self.event_queue_watermarks = event_queue_watermarks
+        self.stats_dump_interval = stats_dump_interval
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, SessionOptions):
+            return False
+        return (
+            self.message_compression_algorithm == other.message_compression_algorithm
+            and self.timeouts == other.timeouts
+            and self.host_health_monitor == other.host_health_monitor
+            and self.num_processing_threads == other.num_processing_threads
+            and self.blob_buffer_size == other.blob_buffer_size
+            and self.channel_high_watermark == other.channel_high_watermark
+            and self.event_queue_watermarks == other.event_queue_watermarks
+            and self.stats_dump_interval == other.stats_dump_interval
+        )
+
+    def __ne__(self, other: object) -> bool:
+        return not self == other
+
+    def __repr__(self) -> str:
+        attrs = (
+            "message_compression_algorithm",
+            "timeouts",
+            "host_health_monitor",
+            "num_processing_threads",
+            "blob_buffer_size",
+            "channel_high_watermark",
+            "event_queue_watermarks",
+            "stats_dump_interval",
+        )
+
+        params = []
+        for attr in attrs:
+            value = getattr(self, attr)
+            if value is not None:
+                params.append(f"{attr}={value!r}")
+
+        return "SessionOptions(%s)" % ", ".join(params)
```

Comment:
```suggestion
        return f"SessionOptions({', '.join(params)})"
```

Reply:
Applying this change more broadly, as this idiom is used in `Timeouts` and the existing `QueueOptions` class

Reply:
Fixed in commit da487a05e71601d206a4e0d7a420788d959c8fd1.